src/client.js

```diff
@@ -26,7 +27,13 @@ function getMDXComponent(code, globals) {
  *
  */
 function getMDXExport(code, globals) {
-  const scope = {React, ReactDOM, _jsx_runtime, ...globals}
+  const scope = {
+    React,
+    ReactDOM,
+    _jsx_runtime:
+      process.env.NODE_ENV === 'production' ? _jsx_runtime : _jsx_dev_runtime,
```

Comment:
This is likely moot since you ship CJS which bundlers have a hard time tree-shaking. But once it's shipped as ESM, bundlers will be able to DCE `react/jsx-dev-runtime` in prod